lib/Dialect/FIRRTL/Transforms/ExpandWhens.cpp

```diff
@@ -126,17 +126,13 @@ class LastConnectResolver : public FIRRTLVisitor<ConcreteT> {
   /// Get the destination value from a connection.  This supports any operation
   /// which is capable of driving a value.
   static Value getDestinationValue(Operation *op) {
-    if (auto connect = dyn_cast<ConnectOp>(op))
-      return connect.dest();
-    return cast<StrictConnectOp>(op).dest();
+    return cast<FConnectLike>(op).dest();
   }
 
   /// Get the source value from a connection. This supports any operation which
   /// is capable of driving a value.
   static Value getConnectedValue(Operation *op) {
-    if (auto connect = dyn_cast<ConnectOp>(op))
```

Comment:
.getOperand(0) would work just as well.

Reply:
I would keep the original one since it would be more concise and easier to refactor in the future.